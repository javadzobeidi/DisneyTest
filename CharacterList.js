/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

 import React,{useState} from 'react';
 import {getColumnNumber} from './Utils/General'
 import { useQuery } from 'react-query';
 import axios from 'axios';
 import {
   SafeAreaView,
   ScrollView,
   StatusBar,
   StyleSheet,
   Image,
   TouchableOpacity,
   ListView  ,
 FlatList,
   Text,
   useColorScheme,
   View,
 } from 'react-native';
 import {Button} from './Component/Button'
 import {
   Colors,
   DebugInstructions,
   Header,
   LearnMoreLinks,
   ReloadInstructions,
 } from 'react-native/Libraries/NewAppScreen';
 import tw from './Lib/tailwind';
 const data=
 {
   "data": [
     {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     },
     {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     },
     {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     }, {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     }
     , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     },
      {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     } , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     } , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     } , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     } , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     } , {
       "_id":308,
       "films":["Tangled","Tangled: Before Ever After"],
       "shortFilms":["Tangled Ever After","Hare Peace"],
       "tvShows":["Once Upon a Time","Tangled: The Series"],
       "videoGames":["Disney Princess Enchanting Storybooks","Hidden Worlds","Disney Crossy Road","Kingdom Hearts III"],
       "parkAttractions":["Celebrate the Magic","Jingle Bell, Jingle BAM!"],
       "allies":[],
       "enemies":[],
       "sourceUrl":"https://disney.fandom.com/wiki/Queen_Arianna",
       "name":"Queen Arianna",
       "imageUrl":"https://static.wikia.nocookie.net/disney/images/1/15/Arianna_Tangled.jpg/revision/latest?cb=20160715191802",
       "createdAt":"2021-04-12T01:33:34.458Z",
       "updatedAt":"2021-04-12T01:33:34.458Z",
       "url":"https://api.disneyapi.dev/characters/308",
       "__v":0
     }
   ],
   "count": 50,
   "previousPage": "https://api.disneyapi.dev/characters?page=3",
   "nextPage": "https://api.disneyapi.dev/characters?page=5",
 }
 
 
 const CharacterList = () => {
 
   const fetchPosts = async () => {
     const { data } = await axios.get('https://api.disneyapi.dev/characters');
     return data;
   }
   const usePosts = () => useQuery('posts', fetchPosts);
 
 
   const { data, isLoading, isSuccess } = usePosts();
 
 
   const renderItem = ({ item }) => (
     <CharacterListItem data={item} />
   );
 
 
 
     return (
         <SafeAreaView style={tw`bg-white h-full   `} >
       <ScrollView style={tw`my-5 px-5  `}>
         
       <Text style={tw`text-2xl text-black font-bold my-5 `}>List of Characters</Text>
      <View style={tw`w-full`} >
  
      <FlatList
        horizontal={false}
      columnWrapperStyle={{flex:1, justifyContent: 'space-between'}}
   data={data.data}
   numColumns={2}
   keyExtractor={(_item, index) => index.toString()}
   renderItem={({ item }) => (
     <CharacterListItem
       data={item}
     />
   )}
 />
 <View style={tw`flex-1 flex-row`}>
 <Button className='flex-1 mr-4 h-16' title='Show 1' />
 <Button className='flex-1 mr-4 h-16' title='Show 2' />
 </View>
      </View>
      </ScrollView>
         </SafeAreaView>
     );
 };
 
 
 
 
  const CharacterListItem = ({
    data
  }) => (
   <View style={tw`mr-4 flex-1  text-center items-center justify-center 	rounded-lg mb-4 	py-2 	 bg-background_primary ` } >
   <View style={tw`w-20 h-20 rounded-full bg-indigo-50`}>
       <Image
         style={tw`w-28 h-28 rounded-full`}
         source={data.imageUrl }
       />
     </View>
          <Text style={tw`text-white pt-4`}>{data.name}</Text>
          <View style={tw`w-full px-4 pt-4`}>
     <Button title='Show More' />
     </View>
        </View>
  )
 
 
  const RightArrow=()=>{
    return (
     <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M17 8l4 4m0 0l-4 4m4-4H3"></path></svg>
    )
  }
 
 export default CharacterList;
 