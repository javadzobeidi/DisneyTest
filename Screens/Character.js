/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, { useState, useEffect } from 'react';
import Icon from 'react-native-vector-icons/AntDesign';
import {
  ScrollView,
  Image,
  View,
} from 'react-native';

import tw from '../Lib/tailwind';
import { Text ,Button} from '../Component'
import { useNavigation } from '@react-navigation/native';


export function Character({ props }) {
  const navigation = useNavigation();
console.log(props);
  return (
    <View style={tw`flex-1`}>
     
        <ScrollView style={tw` bg-background_primary mx-4 my-4`}>
      <View style={tw` mr-4 flex-col   mx-2 my-6 flex-row text-center  items-start  	rounded-lg mb-4 	py-6 	 `} >
        <View style={tw`flex-2 px-3    `}>
          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Name:</Text>
            <Text className=' flex-1'>{props.name}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Films:</Text>
            <Text className='flex-1'>{props.films}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Short Films :</Text>
            <Text className='flex-1'>{props.shortFilms}</Text>
          </View>

          <View style={tw`flex-row py-2 text-justify  flex-1 	`}>
            <Text className=' flex-1 font-bold'>Video Games :</Text>
            <Text className='flex-1'>{props.videoGames}</Text>
          </View>

 
        </View>

        <View style={tw`flex-1 py-4  items-start	justify-start	`}>
          <View style={tw`  rounded-full bg-indigo-50`}>
            <Image
              style={tw`w-28 h-28 rounded-full`}
              source={{uri:props.imageUrl}}

            />
          </View>

        </View>

      </View>
      
      </ScrollView>
     

      <View style={tw`w-34  px-4 py-4  `}>
      
<Button className='flex flex-row 	' onPress={()=>navigation.navigate('characterList')} title='Back' >
<Icon name="arrowleft" size={20} color={'white'} style={tw`mx-4`}/>
  </Button>
  
</View>

     
      </View>

  );
};

