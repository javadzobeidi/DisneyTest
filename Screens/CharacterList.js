/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

 import React,{useState,useEffect} from 'react';
 import { useQuery } from 'react-query';
 import Icon from 'react-native-vector-icons/AntDesign';

 import {
   ScrollView,
  
   Image,
   FlatList,
   Text,
   Alert ,
   View,
 } from 'react-native';
 import {Button,LoadingLayout} from '../Component'
 import tw from '../Lib/tailwind';
 import {GetCharacterList} from '../Api'
 
 export const CharacterList = ({ navigation }) => {

    const [page,setPage]=useState(1);
    const [response,setReponse]=useState([]);
    const [error,setError]=useState(null);

 const {  isLoading, isSuccess ,data,
    isFetching,
    isRefetching,
    isPreviousData,
} = useQuery(['posts',page], ()=>GetCharacterList(page),{onSuccess ,cacheTime: 0 });





function onSuccess(result) {
  if (result.httpCode==200)
  {
    setReponse(result.response)
  }
  else
  {
    Alert.alert(
      result.message
   );

    //setError(result.message);
  }

}



   const renderItem = ({ item }) => (
     <CharacterListItem data={item} />
   );
 
     return (
        <View>
{(isLoading ||isRefetching)  && (
                   <LoadingLayout />
            )}

          {error && <View stlye={tw`flex`}><Text>{error}</Text>
          

          </View>}

{response &&  (
        <ScrollView  nestedScrollEnabled={true} style={tw`my-5 pl-4    `}>
       <Text style={tw`text-2xl text-black font-bold my-5 `}>List of Characters</Text>
      <View style={tw`w-full`} >
  
      <FlatList
      inverted
        horizontal={false}
      columnWrapperStyle={{flex:1, justifyContent: 'space-between'}}
   data={response.data}
   numColumns={2}
   keyExtractor={(_item, index) => index.toString()}
   renderItem={({ item }) => (
     <CharacterListItem
       data={item} navigation={navigation}
     />
   )}
 />
 <View style={tw`flex-1 flex-row pr-4`}>

  
 <Button className='flex-1 flex flex-row  justify-around  mr-4 h-16  ' disabled={response.previousPage=== undefined} onPress={()=>setPage(page-1)}  >
 <Icon name="arrowleft" size={25} color={'white'} style={tw``}/>
 <Text style={tw`font-bold text-lg	  text-white`}>Back</Text>
   </Button>

     <Button className='flex-1 flex  flex-row justify-around	    h-16' disabled={!response.nextPage} onPress={()=>setPage(page+1)}  >
 <Text style={tw`font-bold text-lg	  text-white`}>Next</Text>
 <Icon name="arrowright" size={25} color={'white'} style={tw``}/>
   </Button> 


 </View>
      </View>
      </ScrollView>
)}
      </View>
        
     );
 };


  const CharacterListItem = ({
    navigation,
    data
  }) => (
   <View style={tw`mr-4 flex-1  text-center items-center justify-center 	rounded-lg mb-4 	py-2 	 bg-background_primary ` } >
   <View style={tw`w-20 h-20 rounded-full bg-indigo-50`}>
       <Image
         style={tw`w-full h-full rounded-full`}
         source={{uri:data.imageUrl}}
         />
     </View>
          <Text style={tw`text-white pt-4`}>{data.name}</Text>
          <View style={tw`w-full px-4 pt-4`}>

     <Button onPress={()=>navigation.navigate('character',{id:data._id})} title='Show More' />
     </View>
        </View>
  )
 
 
  const RightArrow=()=>{
    return (
     <svg class="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M17 8l4 4m0 0l-4 4m4-4H3"></path></svg>
    )
  }
 
 