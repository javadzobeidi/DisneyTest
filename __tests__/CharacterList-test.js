import React from 'react';
import renderer from 'react-test-renderer';
import {CharacterList} from '../Screens/CharacterList'
import { QueryClient, QueryClientProvider } from 'react-query';
const queryClient = new QueryClient();

jest.useFakeTimers();

test('renders correctly', () => {
  const tree = renderer.create(
    <QueryClientProvider client={queryClient}>
    <CharacterList />
</QueryClientProvider>
).toJSON();
  expect(tree).toMatchSnapshot();
});