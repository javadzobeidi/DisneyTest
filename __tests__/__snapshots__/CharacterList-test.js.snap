// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F51CFF",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 100,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "flex": 1,
            "left": 0,
            "opacity": 0.9,
            "position": "absolute",
            "top": 0,
            "width": 750,
            "zIndex": 100,
          },
          Object {
            "height": 1334,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "zIndex": 100,
          }
        }
      >
        <ActivityIndicator
          color="#0000ff"
          size="large"
        />
        <Text>
          Loading
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    nestedScrollEnabled={true}
    style={
      Object {
        "marginBottom": 20,
        "marginTop": 20,
        "paddingLeft": 16,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 24,
            "fontWeight": "bold",
            "lineHeight": 32,
            "marginBottom": 20,
            "marginTop": 20,
          }
        }
      >
        List of Characters
      </Text>
      <View
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          invertStickyHeaders={true}
          inverted={true}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
              undefined,
            ]
          }
          viewabilityConfigCallbackPairs={Array []}
        >
          <View />
        </RCTScrollView>
        <View
          style={
            Object {
              "flexBasis": "0%",
              "flexDirection": "row",
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": "disabled",
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#c0cf80",
                "borderRadius": 8,
                "display": "flex",
                "elevation": 3,
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 64,
                "justifyContent": "space-around",
                "marginRight": 16,
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.125,
                "shadowRadius": 3,
                "width": "100%",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                  },
                  Object {},
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "lineHeight": 28,
                }
              }
            >
              Back
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": "disabled",
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#c0cf80",
                "borderRadius": 8,
                "display": "flex",
                "elevation": 3,
                "flexBasis": "0%",
                "flexDirection": "row",
                "flexGrow": 1,
                "flexShrink": 1,
                "height": 64,
                "justifyContent": "space-around",
                "opacity": 1,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 1,
                  "width": 1,
                },
                "shadowOpacity": 0.125,
                "shadowRadius": 3,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "lineHeight": 28,
                }
              }
            >
              Next
            </Text>
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 25,
                  },
                  Object {},
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
